astropy/wcs/wcs.py

```diff
@@ -1175,7 +1193,64 @@ def write_dist(num, cpdis):
         write_dist(1, self.cpdis1)
         write_dist(2, self.cpdis2)
 
-    def _remove_sip_kw(self, header):
+    def _fix_pre2012_scamp_tpv(self, header, wcskey=""):
+        """
+        Replace -TAN with TPV (for pre-2012 SCAMP headers that use -TAN
+        in CTYPE). Ignore SIP if present. This follows recommendations in
+        Section 7 in
+        http://web.ipac.caltech.edu/staff/shupe/reprints/SIP_to_PV_SPIE2012.pdf.
+
+        This is to deal with pre-2012 headers that may contain TPV with a
+        CTYPE that ends in '-TAN' (post-2012 they should end in '-TPV' when
+        SCAMP has adopted the new TPV convention).
+        """
+        if isinstance(header, (str, bytes)):
+            return
+
+        wcskey = wcskey.strip().upper()
+        cntype = [
+            (nax, header.get(f"CTYPE{nax}{wcskey}", "").strip())
+            for nax in range(1, self.naxis + 1)
+        ]
+
+        tan_axes = [ct[0] for ct in cntype if ct[1].endswith("-TAN")]
+
+        if len(tan_axes) == 2:
+            # check if PVi_j with j >= 5 is present and if so, do not load SIP
+            tan_to_tpv = False
+            for nax in tan_axes:
+                js = []
+                for p in header[f"PV{nax}_*{wcskey}"].keys():
+                    prefix = f"PV{nax}_"
+                    if p.startswith(prefix):
+                        p = p[len(prefix) :]
+                        p = p.rstrip(wcskey)
+                        try:
+                            p = int(p)
+                        except ValueError:
+                            continue
```

Comment:
I think you'd want to just `return` here - if this fails, the header is clearly a mess so best not to even try fixing anything.

Reply:
Not really. `PV1_A11` might be a very valid custom keyword used by users. We do not that. There are many keywords that look similar - we do not abort code on them.

Reply:
OK, that makes sense. Then it should be fine as is!